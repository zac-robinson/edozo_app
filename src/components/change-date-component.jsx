import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import {
  incrementDateIndex,
  decrementDateIndex
} from '../actions/common-actions';
import { ButtonGroup, DateButton } from '../styles/change-date.styles';

export class ChangeDate extends React.Component {
  render() {
    return (
      <ButtonGroup>
        <DateButton onClick={this.props.decrementDateIndex}>
          Previous Day
        </DateButton>
        <DateButton onClick={this.props.incrementDateIndex}>
          Next Day
        </DateButton>
      </ButtonGroup>
    );
  }
}

ChangeDate.propTypes = {
  incrementDateIndex: PropTypes.func.isRequired,
  decrementDateIndex: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => {
  return {
    incrementDateIndex: () => dispatch(incrementDateIndex()),
    decrementDateIndex: () => dispatch(decrementDateIndex())
  };
};

const mapStateToProps = ({ selectedCity, forecastDataByCity, dateIndex }) => {
  return {
    dateIndex,
    numDays: Object.keys(forecastDataByCity[selectedCity]).length
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ChangeDate);
