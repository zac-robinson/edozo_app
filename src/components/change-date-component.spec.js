import React from 'react';
import { shallow } from 'enzyme';

import { ChangeDate } from './change-date-component';

describe('ChangeDate component', () => {
  it('should render correctly', () => {
    const props = {
      incrementDateIndex: () => {},
      decrementDateIndex: () => {}
    };
    expect(shallow(<ChangeDate {...props} />)).toMatchSnapshot();
  });
});
