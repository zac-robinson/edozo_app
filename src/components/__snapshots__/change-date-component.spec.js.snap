// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeDate component should render correctly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    Previous Day
  </styled.button>
  <styled.button
    onClick={[Function]}
  >
    Next Day
  </styled.button>
</styled.div>
`;
